dex/src/main/scala/com/wavesplatform/dex/actors/address/AddressActor.scala

```diff
@@ -375,25 +375,32 @@ class AddressActor(
       } else {
         log.debug(s"$command, to cancel: ${toCancelIds.mkString(", ")}")
         runWithIgnoredSpan {
-          context.actorOf(BatchOrderCancelActor.props(toCancelIds.toSet, command.source, self, sender(), settings.batchCancelTimeout))
+          context.actorOf(BatchOrderCancelActor.props(
+            toCancelIds.map(id => id -> AddressActor.Event.OrderCanceled(id).asRight).to(ListMap),
+            command.source,
+            self,
+            sender(),
+            settings.batchCancelTimeout
+          ))
         }
       }
 
     case command: Command.CancelOrders =>
       val allActiveOrderIds = getActiveLimitOrders(None).map(_.order.id()).toSet
-      val toCancelIds = allActiveOrderIds.intersect(command.orderIds)
-      val unknownIds = command.orderIds -- allActiveOrderIds
+      val toCancelIds = command.orderIds.filter(allActiveOrderIds.contains)
 
-      log.debug(
-        s"$command, total orders: ${allActiveOrderIds.size}, to cancel (${toCancelIds.size}): ${toCancelIds
-          .mkString(", ")}, unknown ids (${unknownIds.size}): ${unknownIds.mkString(", ")}"
-      )
+      val response = command.orderIds.map { id =>
+        id -> (if (toCancelIds.contains(id)) AddressActor.Event.OrderCanceled(id).asRight
+               else error.OrderNotFound(id).asLeft[AddressActor.Event.OrderCanceled])
+      }.to(ListMap)
 
-      val initResponse = unknownIds.map(id => id -> error.OrderNotFound(id).asLeft[AddressActor.Event.OrderCanceled]).toMap
-      if (toCancelIds.isEmpty) sender() ! Event.BatchCancelCompleted(initResponse)
+      if (toCancelIds.isEmpty)
+        sender() ! Event.BatchCancelCompleted(command.orderIds.map(id =>
+          id -> error.OrderNotFound(id).asLeft[AddressActor.Event.OrderCanceled]
+        ).toMap)
```

Comment:
Why can't we use `response` here? If `toCancelIds` is empty, `response` will contain only `OrderNotFound`, isn't it?

Reply:
fixed